src/main/java/org/spongepowered/api/data/property/DoubleProperty.java

```diff
@@ -24,53 +24,9 @@
  */
 package org.spongepowered.api.data.property;
 
-import org.spongepowered.api.data.Property;
-import org.spongepowered.api.data.property.item.FoodRestorationProperty;
-import org.spongepowered.api.data.property.item.SaturationProperty;
-import org.spongepowered.api.util.Coerce;
-
 /**
- * Represents an item property that has an integer value. Examples may include
- * {@link FoodRestorationProperty}, {@link SaturationProperty} etc.
+ * A property type for {@link Double} values.
  */
-public class DoubleProperty extends AbstractProperty<String, Double> {
-
-    /**
-     * Create a new integer property with the specified value.
-     *
-     * @param value value to match
-     */
-    public DoubleProperty(double value) {
-        super(Coerce.toDouble(value));
-    }
-
-    /**
-     * Create a new integer property with the specified value and logical
-     * operator.
-     *
-     * @param value value to match
-     * @param operator logical operator to use when comparing to other
-     *      properties
-     */
-    public DoubleProperty(double value, Operator operator) {
-        super(value, operator);
-    }
-
-    /**
-     * Create a new integer property with the specified value and logical
-     * operator.
-     *
-     * @param value value to match
-     * @param operator logical operator to use when comparing to other
-     *      properties
-     */
-    public DoubleProperty(Object value, Operator operator) {
-        super(Coerce.toDouble(value), operator);
-    }
-
-    @Override
-    public int compareTo(Property<?, ?> other) {
-        return this.getValue().compareTo(other == null ? 1 : Coerce.toDouble(other.getValue()));
-    }
+public interface DoubleProperty extends Property<Double> {
```

Comment:
If you're exposing `OptionalInt`s, we should consider making these return their primitive type values as well.

Reply:
Something like:
`int getProperty(IntProperty property, int defaultValue)`?

Reply:
Then I can maybe get rid of the `OptionalInt` and `OptionalDouble` methods.